grails-app/taglib/grails/plugin/sentry/SentryTagLib.groovy

```diff
@@ -0,0 +1,47 @@
+package grails.plugin.sentry
+
+import io.sentry.Sentry
+import io.sentry.SentryTraceHeader
+
+class SentryTagLib {
+
+    static final namespace = 'sentry'
+
+    SentryConfig sentryPluginConfig
+
+    /**
+     * Meta tag containing sentry trace.
+     *
+     * To enable <a href="https://docs.sentry.io/product/sentry-basics/tracing/distributed-tracing/">distributed tracing</a>
+     * you need to <a href="https://docs.sentry.io/platforms/java/performance/connect-services/">connect services</a> (frontend)
+     * by adding this tag (`<sentry:traceMeta />`) to `<head>` of your gsp view.
+     */
+    def traceMeta = { attrs, body ->
+        SentryTraceHeader sentryTrace = Sentry.currentHub.span?.toSentryTrace()
+        if (sentryTrace) {
+            out << "<meta name=\"${SentryTraceHeader.SENTRY_TRACE_HEADER}\" content=\"${sentryTrace.value}\" />"
+        }
+    }
+
+    /**
+     * Link to current trace in sentry app.
+     *
+     * If `linkPrefix` is configured and current trace is sampled then this tag displays link composed of `linkPrefix`
+     * and current trace id.
+     *
+     * Link label can by provided in tag's body otherwise embedded svg sentry logo is displayed.
+     */
+    def traceLink = { attrs, body ->
+        SentryTraceHeader sentryTrace = Sentry.currentHub.span?.toSentryTrace()
+        String linkPrefix = sentryPluginConfig.linkPrefix
+        if (sentryTrace?.sampled && linkPrefix) {
+            out << """<a href="${linkPrefix}/performance/trace/${sentryTrace.traceId}" target="_blank class="sentry-trace-link">"""
+            out << (body() ?: """
+                <svg style="vertical-align: text-bottom; margin: 0 8px;" fill="currentColor" viewBox="0 0 16 16" height="16px" width="16px">
```

Comment:
What does this svg do and why do we need to display it?

Reply:
It's just convenience feature that you can display link to current trace in sentry app and if you don't provide body of link, sentry logo is displayed - it's also mentioned in javadoc.

Reply:
OK